android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/PhysicalWebBroadcastService.java

```diff
@@ -98,40 +102,58 @@ public void onCreate() {
 
     @Override
     public int onStartCommand(Intent intent, int flags, int startId) {
-        Log.d(TAG, "SERVICE onStartCommand");
-        IntentFilter filter = new IntentFilter(BluetoothAdapter.ACTION_STATE_CHANGED);
-        registerReceiver(mReceiver, filter);
-        mDisplayUrl = intent.getStringExtra(DISPLAY_URL_KEY);
-        Log.d(TAG, mDisplayUrl);
-
-        if (hasValidUrlLength(mDisplayUrl) && checkAndHandleAsciiUrl(mDisplayUrl)) {
-            // Set the url if we can
-            Log.d(TAG, "valid length");
-            mShareUrl = mDisplayUrl;
-            broadcastUrl();
+        mStartedByRestart = (intent == null);
+        fetchBroadcastData(intent);
+        if (mDisplayUrl != null){
```

Comment:
I'd prefer you to just do:
    if (mDisplay == null) {
      stopSelf();
      return START_STICKY;
    }

then there'd be less levels of indentation.